src/QVContainer.tsx

```diff
@@ -1,87 +1,272 @@
 import * as React from "react";
-import * as ReactDOM from "react-dom";
 
-import * as QV from './QVClient';
-
-import TRCContext from "trc-react/dist/context/TRCContext";
 import * as XC from "trc-httpshim/xclient";
 
+import * as QV from "./QVClient";
+import { Loader } from "./Loader";
+
 export interface IQVState {
-    QVClient: QV.QVClient;
-    SheetId: string;
-    Model: QV.IQVModel; // set on Init 
-  }
-  
+  AuthToken: string;
+  QVClient: QV.QVClient;
+  SheetId: string;
+  Model: QV.IQVModel; // set on Init
+}
+
 // Passed from Plugin hosting interface to call SetSheetRef()
 interface IPluginSheetRef {
-    Server: string;
-    AuthToken: string;
-    SheetId: string;
+  Server: string;
+  AuthToken: string;
+  SheetId: string;
 }
 
-// Hack since Context doesn't work. 
-declare var _contextHack: IQVState;
-  
-
-// Replaces <SheetContainer>
-// https://github.com/Voter-Science/trc-react/blob/master/src/SheetContainer.tsx
-// Sets a context, which can be access in a child component by "this.context"
-export class QVContainer extends React.Component<{}, IQVState>
-{
-  public constructor(props: any) {
-    super(props);
-
-    // Ordering:
-    // - DOM has an html element
-    // - <SheetContainer> is rendered to that element. This sets the global
-    // - Load complete.
-    // - PluginMain() is called (after the OnLoad event). This reads the global
-    //     to get <SheetContainer> and call setSheetRef();
-    var x: any = window;
-    x.mainMajor = this;
+function arrayMove(arr: any[], oldIndex: number, newIndex: number) {
+  if (newIndex >= arr.length) {
+    let k = newIndex - arr.length + 1;
+    while (k--) {
+      arr.push(undefined);
+    }
   }
-  render() {
-    _contextHack = this.state;
+  arr.splice(newIndex, 0, arr.splice(oldIndex, 1)[0]);
+  return arr;
+}
+
+export function withQVContainer<P extends object>(
+  WrappedComponent: React.ComponentType<P>
+) {
+  return class QVContainer extends React.Component<{}, IQVState> {
```

Comment:
QVContainer should be similar to SheetContainer in that it's just a generic shell to get the auth token (vai the setSheetRef call made by the host) and then sets the QVClient in the context.  Other QV plugins may use QVContainer. 
The editing then should be in its own react control.